### DISCUSSION TITLE
Issue with Setting Up Local Repository in Weblate

### DISCUSSION THREAD

**Discussion by cankirca** [WeblateOrg/weblate#11856] [Q&A]
Hi,

I'm encountering an issue while trying to set up a local repository in Weblate. Here's a detailed description of the problem and the steps I’ve taken so far:

**Context:**
Yesterday, I successfully added a project with a component and uploaded translations in three different languages without any issues. Today, however, when I tried to create a new translation component within the same project, Weblate requires me to enter a "Source code repository" value, which is now a mandatory field. It didn't accept anything for local.

**Example URL:**
http://domain.com/create/component/?project=5&category=3

**Observations:**
- Old components show 'local:' as the "Source code repository" value and 'main' as the branch.
- I am unable to use 'local:' as the value for new components.
- Also, the dropdown options available are Gerrit, Git, Git with force push, Mercurial, and Subversion, but there is no option for "No remote repository" as mentioned in the help text below the dropdown: "Choose No remote repository for local: URL."

**Questions:**
1. Why is the "Source code repository" value now mandatory when it wasn’t required previously?
2. How can I set up a local repository without specifying a remote repository?

**Steps to Reproduce:**
1. Navigate to the component creation URL: `http://domain.com/create/component/?project=5&category=3`.
2. Attempt to create a new translation component within an existing project.
3. Observe that the "Source code repository" field is mandatory and 'local:' is not accepted as a value.

**Possible Workarounds:**
- If you have any insights on why this change occurred or how to revert to the previous setup where 'local:' was accepted, please share.
- Any configuration changes or additional steps that need to be taken to allow for local repositories without remote URLs would be highly appreciated.

Thank you in advance for your assistance. Looking forward to your expert advice.

**Comment by cankirca** [comment]
Hi again,
 
I've created a test project on Hosted Weblate, local repo field is mandatory in there as well and I can't create component. What should I enter as data to this field?
https://hosted.weblate.org/create/component/vcs/
BTW. I can see No remote repository option in the component creation screen at http://domain.com/admin/trans/component/add/
bu tnot listed in en uder UI.

**Comment by nijel** [comment]
“From version control” needs existing remote version control, choose “Upload translations files”, “Translate document”, or “Start from scratch” to create a component without a remote repository.

**Comment by cankirca** [comment]
Hi @nijel,
 Thank you very much for your answer, I have a vision issue and can't noticed other tabs on the screen :D Now it worked!
 Could you take a look the website URL issue at https://github.com/WeblateOrg/weblate/discussions/11830#discussioncomment-9766679 please?
 Regards!
